### ISSUE TITLE
Prior dependence on Ne

### ISSUE THREAD

**Issue by jelliotthiggins** [tskit-dev/tsdate#163]
Hello!
In using tsdate, my PI Dr. Swamidass and I noticed a pretty heavy dependency on Ne in dating trees, presumably due to how it is weighted in the prior. Implementing the example from the tsdate tutorial with varying Ne shows that there's a strong positive correlation between the effective population and the age of the oldest node (this age is the cleanest example, but the entire distribution of ages in the tree sequence shifts back in time with increased Ne). 

```
import msprime, tsinfer, tsdate

for Ne in [10000, 20000, 50000, 100000]:
    sample_ts = msprime.simulate(sample_size=10, Ne=Ne,
                                            length=1e4,
                                            mutation_rate=1e-8,
                                            recombination_rate=1e-8,
                                            random_seed=2)
    sample_data = tsinfer.SampleData.from_tree_sequence(sample_ts)
    inferred_ts = tsinfer.infer(sample_data).simplify(keep_unary=False)
    dated_ts = tsdate.date(inferred_ts, Ne=Ne, mutation_rate=1e-8)
    max_age = max([dated_ts.node(tree.root).time for tree in dated_ts.trees()])
    print(f"Ne={Ne}, oldest node is {max_age} generations")
```

yields these results:

```
Ne=10000, oldest node is 58642.110459065596 generations
Ne=20000, oldest node is 68177.5390732491 generations
Ne=50000, oldest node is 179238.90804494536 generations
Ne=100000, oldest node is 475261.05302381696 generations
```

Is this something that you've noticed already?

Thank you!
Jack Elliott-Higgins

**Comment by swamidass** [comment]
Jack, do you get the same results when you take the simulation outside the loop, with a single fixed Ne? That might be the issue.

**Comment by swamidass** [comment]
Okay, I was able to reproduce the issue with the correct code, this time using a fixed Ne and large genome size for the simulation.

```

This produces results which seemsed
sample_ts = msprime.simulate(sample_size=10, Ne=10000,
                                            length=1e6,
                                            mutation_rate=1e-8,
                                            recombination_rate=1e-8,
                                            random_seed=2)
sample_data = tsinfer.SampleData.from_tree_sequence(sample_ts)
inferred_ts = tsinfer.infer(sample_data).simplify(keep_unary=False)


for Ne in [10000, 20000, 50000, 100000]:
    dated_ts = tsdate.date(inferred_ts, Ne=Ne, mutation_rate=1e-8)
    max_age = max([dated_ts.node(tree.root).time for tree in dated_ts.trees()])
    
    print("Ne= %6.0f, -> %0.0f generations" % (Ne, max_age))
```

It produces this output, which is approximately correct when Ne is the right value, but increases too much as you increase the Ne from the simulation. 

```
Ne=  10000, -> 52276 generations
Ne=  20000, -> 93114 generations
Ne=  50000, -> 170513 generations
Ne= 100000, -> 242875 generations

```

Is there a way to reduce the strength of the prior to eliminate this problem?

**Comment by hyanwong** [comment]
Hi @jelliotthiggins and @swamidass  - great that you are using the software. Thanks for reporting back. You are right that the only effect of Ne is to change the prior: this should be clearer when https://github.com/tskit-dev/tsdate/pull/156 is merged. We should then state in the docs that `Ne` is only used if no prior is specified using `tsdate.date(... priors=XXX)`. There is currently no way to decrease the strength of the prior, I'm afraid (I don't even know how statistically valid this would be: do you have a proposal for a reasonable way to do it?). However, after #156 is merged, you could create your own prior as a `NodeGridValues` object and pass that in instead? Of course, as the mutation rate goes up (or the recombination rate goes down) then the prior should start being overwhelmed by the data, but with the parameters you are using (rho ~= mu) then I guess the prior is likely to be quite influential.

@awohns might be able to add more detail. If you have the expertise to help improve tsdate, then any input would be most appreciated, I'm sure.

**Comment by awohns** [comment]
Hi Jack and Dr. Swamidass,
Thanks very much for this. The specified `Ne` is only used as a scaling factor in the prior, so misspecification of Ne will cause the resulting estimated values to be less accurate. However, this is only due to scaling in the prior, when the data is strong it will overwhelm the influence of a mispecified `Ne`. For example:



@swamidass, in your previous example I belive the line:
`mean_age = np.mean([sample_ts.node(tree.root).time for tree in dated_ts.trees()])`
should be:
`mean_age = np.mean([sample_ts.node(tree.root).time for tree in sample_ts.trees()])`

Once that's corrected, the average value of the tmrca by tree is 41929, which isn't so far off of what `tsinfer` and `tsdate` estimate. 

As an aside, I'd be interested to hear about your applications of `tsdate` and `tsinfer`!

**Comment by awohns** [comment]
Hi @jelliotthiggins and @swamidass,
Thanks very much for this. Building on what @hyanwong wrote, the specified Ne is only used as a scaling factor in the prior, so misspecification of Ne will cause the resulting estimated values to be less accurate. However, as this is only due to scaling the prior, when the data is strong it will overwhelm the influence of a mispecified Ne. For example:

`import msprime, tsinfer, tsdate
import matplotlib.pyplot as plt
sample_ts = msprime.simulate(sample_size=10, Ne=10000,
                                            length=1e6,
                                            mutation_rate=1e-8,
                                            recombination_rate=1e-8,
                                            random_seed=2)
for Ne in [5000, 10000, 20000, 50000, 100000]:
    dated_ts = tsdate.date(sample_ts, Ne, mutation_rate=1e-8)
    plt.scatter(sample_ts.tables.nodes.time[sample_ts.tables.mutations.node],
                dated_ts.tables.nodes.time[dated_ts.tables.mutations.node], s=0.5, label="Ne={}".format(Ne))
    plt.yscale("log")
    plt.xscale("log")
    plt.xlim(10, 1000000)
    plt.ylim(10, 1000000)
    plt.plot(plt.xlim(), plt.ylim())
plt.legend()`
![image](https://user-images.githubusercontent.com/10664424/103040326-fe495200-456a-11eb-8f49-cb7528540342.png)
However, when we set the mutation rate to be 1e-6 and the recombination rate to be 1e-8:
![image](https://user-images.githubusercontent.com/10664424/103040321-fa1d3480-456a-11eb-98da-7386bbe1a4ef.png)


@swamidass, in your previous example I belive the line:
mean_age = np.mean([sample_ts.node(tree.root).time for tree in dated_ts.trees()])
should be:
mean_age = np.mean([sample_ts.node(tree.root).time for tree in sample_ts.trees()])

Once that's corrected, the average value of the tmrca by tree is 41929, which isn't so far off of what tsinfer and tsdate estimate.

As an aside, I'd be interested to hear about your applications of tsdate and tsinfer!

**Comment by swamidass** [comment]
I suppose the issue is that even with a lot of date, the prior is not overwhelmed.

**Comment by hyanwong** [comment]
Sorry - I don't quite understand. I think the plots above show that the conditional coalescent prior *is* overwhelmed by the data from mutation, as the mutation rate (relative to the recombination rate) increases.

Or do you mean that, during inference, the mutation rate itself should be thought of as a prior?

**Comment by hyanwong** [comment]
Can this be closed once #156 is merged?

**Comment by jelliotthiggins** [comment]
@hyanwong Yeah I think that's fine! Having the ability to pass in a prior will help. 

Just for reference, we had been running tsdate on simulated data using a mutation to recombination ratio of 2:1 (fairly realistic), so I filled in some data points between equality and 100:1 to demonstrate how that looks. The prior's dependence on Ne is definitely still very noticeable, but allowing the user to pass in a prior should be a way around it at least. 

![image](https://user-images.githubusercontent.com/26074642/104491183-e5e3ba80-559f-11eb-920c-0fe6edcc2e75.png)

![image](https://user-images.githubusercontent.com/26074642/104491206-ed0ac880-559f-11eb-9789-860f0c202993.png)
